packages/flutter/test/html/selectable_region_context_menu_test.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: undefined_class, undefined_getter, undefined_setter
+
+import 'dart:html' as html;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  html.Element? element;
+  final RegisterViewFactory originalFactory = PlatformSelectableRegionContextMenu.registerViewFactory;
+  PlatformSelectableRegionContextMenu.registerViewFactory = (String viewType, Object Function(int viewId) fn, {bool isVisible = true}) {
+    element = fn(0) as html.Element;
+    html.document.body!.append(element!);
```

Comment:
Why does it need to be attached? Is it to be able to programmatically dispatch events? If so, consider adding a comment explaining it.

Reply:
Also, I wonder if the engine should not ignore requests to create platform views. If core framework widgets will be creating platform views, perhaps they should really be attached as they normally would so we could write tests that capture everything end-to-end? This would be outside the scope of this PR though, as that would require a change in the engine.

Reply:
added comment. I agree that may be a better way to test this kind of code